addon/services/engine-router-service.js

```diff
@@ -0,0 +1,118 @@
+import Service from '@ember/service';
+import { assert } from '@ember/debug';
+import { action, computed } from '@ember/object';
+import { reads } from '@ember/object/computed';
+import { getOwner } from '@ember/application';
+import Evented from '@ember/object/evented';
+import { namespaceEngineRouteName } from '../utils/namespace-engine-route-name';
+import { getRootOwner } from '../utils/owner';
+
+export default class EngineRouterService extends Service.extend(Evented) {
+  init() {
+    super.init();
+
+    this._externalRoutes = getOwner(this)._externalRoutes;
+    this._mountPoint = getOwner(this).mountPoint;
+    this.rootApplication = getRootOwner(this);
+
+    this.externalRouter.on('routeWillChange', this.onRouteWillChange);
+    this.externalRouter.on('routeDidChange', this.onRouteDidChange);
+  }
+
+  destroy() {
+    this.externalRouter.off('routeWillChange', this.onRouteWillChange);
+    this.externalRouter.off('routeDidChange', this.onRouteDidChange);
+
+    super.destroy();
+  }
+
+  @action
+  onRouteWillChange(...args) {
+    this.trigger('routeWillChange', ...args);
+  }
+
+  @action
+  onRouteDidChange(...args) {
+    this.trigger('routeDidChange', ...args);
```

Comment:
Should these be translated as they will contain the application-wide names instead of the engine-scoped ones (if I'm not mistaken)?